src/Components/Header/Menu/Search.jsx

```diff
@@ -0,0 +1,35 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux';
+import { filterByName } from '../../../store/actions/actions';
+import { Icon, Menu } from 'semantic-ui-react';
+import styled from 'styled-components';
+
+class Search extends Component {
+    state = {
+        active: false
+    }
+    activate = () => this.setState({active: true})
```

Comment:
This is discouraged because it creates a new function every single time this is instantiated. Wastes a ton of memory. Instead do `activate() { .... }`

Reply:
Are you sure? This is still a class method, it's just written using an arrow, so it's an arrow function class method. Supposed to work...